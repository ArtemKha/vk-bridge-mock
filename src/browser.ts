import bridgeMock, { callReceiveOnlyMethod } from './';

// @ts-ignore
window.vkBridgeMock = window.vkConnectMock = bridgeMock;
// @ts-ignore
window.vkBridgeCallReceiveOnlyMethod = callReceiveOnlyMethod;
