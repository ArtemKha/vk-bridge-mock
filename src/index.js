import GetUserInfo from './Handlers/VKWebAppGetUserInfo';
import Init from './Handlers/VKWebAppInit';
import GetAuthToken from './Handlers/VKWebAppGetAuthToken';
import CallAPIMethod from './Handlers/VKWebAppCallAPIMethod';
import GetGeodata from './Handlers/VKWebAppGetGeodata';
import GetPhoneNumber from './Handlers/VKWebAppGetPhoneNumber';
import GetClientVersion from './Handlers/VKWebAppGetClientVersion';
import GetEmail from './Handlers/VKWebAppGetEmail';

export default {
  VKWebAppInit: Init(),
  // Data
  VKWebAppGetAuthToken: GetAuthToken(),
  // Data
  VKWebAppCallAPIMethod: CallAPIMethod(),
  // Data
  VKWebAppGetGeodata: GetGeodata(),
  // Datab
  VKWebAppGetUserInfo: GetUserInfo(),
  // Data
  VKWebAppGetPhoneNumber: GetPhoneNumber(),
  // Data
  VKWebAppGetClientVersion: GetClientVersion(),
  // Data
  VKWebAppGetEmail: GetEmail(),
  // TODO UI
  VKWebAppOpenPayForm: {},
  // TODO UI
  VKWebAppShare: {},
  // TODO UI
  VKWebAppAllowNotifications: {},
  // TODO UI
  VKWebAppDenyNotifications: {},
  // TODO ???
  VKWebAppViewUpdateNavigationState: {},
  // TODO ???
  VKWebAppSetTitle: {},
  // TODO UI
  VKWebAppShowWallPostBox: {},
  // TODO ???
  VKWebAppSetLocation: {},
  // TODO UI
  VKWebAppAllowMessagesFromGroup: {},
  // TODO UI
  VKWebAppJoinGroup: {},
};
