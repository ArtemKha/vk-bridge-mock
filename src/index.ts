import vkConnect, {
  VKConnect,
  VKConnectSubscribeHandler,
  RequestMethodName,
  RequestProps,
  RequestIdProp,
  IOMethodName,
  ReceiveData,
  VKConnectSuccessEvent,
  ReceiveMethodName,
  ReceiveOnlyMethodName,
  VKConnectSend
} from '@vkontakte/vk-connect';
import { mockDataMap } from './mock';

export const receiveOnlyMethods: ReceiveOnlyMethodName[] = [
  'VKWebAppAudioPaused',
  'VKWebAppAudioStopped',
  'VKWebAppAudioTrackChanged',
  'VKWebAppAudioUnpaused',
  'VKWebAppInitAds',
  'VKWebAppLoadAds',
  'VKWebAppUpdateConfig',
  'VKWebAppViewHide',
  'VKWebAppViewRestore'
];

export const ioMethods = Object.keys(mockDataMap).filter(methodName => !receiveOnlyMethods.includes(methodName as any));

const state = {
  listeners: [] as VKConnectSubscribeHandler[],
  currentRequestId: 0,
  getNextRequestId() {
    this.currentRequestId++;

    return this.currentRequestId;
  }
};

const isReceiveMethodExists = (methodName: string): methodName is ReceiveMethodName => methodName in mockDataMap;

const getMockData = <T extends ReceiveMethodName>(
  methodName: T,
  props?: T extends IOMethodName ? RequestProps<T> : {}
): ReceiveData<T> | null => {
  if (isReceiveMethodExists(methodName) && mockDataMap[methodName]) {
    // FIXME
    return mockDataMap[methodName]!(props as any) as ReceiveData<T>;
  }

  return null;
};

const prepareResponse = <K extends ReceiveMethodName>(
  method: K,
  props?: K extends RequestMethodName ? RequestProps<K> & RequestIdProp : RequestIdProp
): VKConnectSuccessEvent<K> | null => {
  if (!isReceiveMethodExists(method)) {
    // TODO
    return null;
  }

  const data = {
    // FIXME: any
    ...getMockData(method, props as any)!,
    request_id: state.getNextRequestId()
  };

  const event: VKConnectSuccessEvent<K> = {
    detail: {
      type: receiveOnlyMethods.includes(method as any) ? method : method + 'Result',
      data
    }
  };

  return event;
};

const broadcastData = (event: VKConnectSuccessEvent<ReceiveMethodName>) => {
  state.listeners.forEach(listener => {
    listener(event);
  });
};

export const callReceiveOnlyMethod = (methodName: ReceiveOnlyMethodName) => {
  if (receiveOnlyMethods.includes(methodName)) {
    const event = prepareResponse(methodName);

    if (event) {
      broadcastData(event);
    }
  }
};

const send: VKConnectSend = async (method, props) => {
  return new Promise((resolve, reject) => {
    if (!isReceiveMethodExists(method)) {
      return;
    }

    const event = prepareResponse(method as ReceiveMethodName, props);

    if (!event) {
      return;
    }

    broadcastData(event);
    resolve(event.detail.data as any);
  });
};

const vkConnectMock: VKConnect = {
  /**
   * Sends an event to the runtime env and returns the Promise object with
   * response data. In the case of Android/iOS application env is the
   * application itself. In the case of the browser, the parent frame in which
   * the event handlers is located.
   *
   * @param method The method (event) name to send.
   * @param [props] Method properties.
   * @returns The Promise object with response data.
   */
  send,

  /**
   * @alias send
   * @deprecated
   */
  sendPromise: send,

  /**
   * Adds an event listener. It will be called any time a data is received.
   *
   * @param listener A callback to be invoked on every event receive.
   */
  subscribe: (fn: VKConnectSubscribeHandler) => void state.listeners.push(fn),

  /**
   * Removes an event listener which has been subscribed for event listening.
   *
   * @param listener A callback to unsubscribe.
   */
  unsubscribe: (fn: VKConnectSubscribeHandler) => {
    const index = state.listeners.indexOf(fn);

    if (index > -1) {
      state.listeners.splice(index, 1);
    }
  },

  /**
   * Checks if a method is supported on runtime platform.
   *
   * @param method Method (event) name to check.
   * @returns Result of checking.
   */
  supports: (method: string): boolean => vkConnect.supports(method),

  /**
   * Checks whether the runtime is a WebView.
   *
   * @returns Result of checking.
   */
  isWebView: (): boolean => vkConnect.isWebView()
};

export default vkConnectMock;
