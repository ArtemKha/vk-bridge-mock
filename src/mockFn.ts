import {
  ReceiveMethodName,
  RequestMethodName,
  IOMethodName,
  RequestProps,
  ReceiveData,
  RequestIdProp,
  VKConnectSuccessEvent,
  VKConnectSubscribeHandler,
  VKConnectSend,
  ReceiveOnlyMethodName
} from '@vkontakte/vk-connect';
import { mockDataMap } from './mockData';

export const receiveOnlyMethods: ReceiveOnlyMethodName[] = [
  'VKWebAppAudioPaused',
  'VKWebAppAudioStopped',
  'VKWebAppAudioTrackChanged',
  'VKWebAppAudioUnpaused',
  'VKWebAppInitAds',
  'VKWebAppLoadAds',
  'VKWebAppUpdateConfig',
  'VKWebAppViewHide',
  'VKWebAppViewRestore'
];

const state = {
  listeners: [] as VKConnectSubscribeHandler[],
  currentRequestId: 0,
  getNextRequestId() {
    this.currentRequestId++;

    return this.currentRequestId;
  }
};

const getMockData = <T extends ReceiveMethodName>(
  methodName: T,
  props?: T extends IOMethodName ? RequestProps<T> : {}
): ReceiveData<T> | null => {
  if (isReceiveMockMethodExists(methodName) && mockDataMap[methodName]) {
    // FIXME
    return mockDataMap[methodName]!(props as any) as ReceiveData<T>;
  }

  return null;
};

export const broadcastData = (event: VKConnectSuccessEvent<ReceiveMethodName>) => {
  state.listeners.forEach(listener => {
    listener(event);
  });
};

export const prepareResponse = <K extends ReceiveMethodName>(
  method: K,
  props?: K extends RequestMethodName ? RequestProps<K> & RequestIdProp : RequestIdProp
): VKConnectSuccessEvent<K> | null => {
  if (!isReceiveMockMethodExists(method)) {
    // TODO
    return null;
  }

  const data = {
    // FIXME: any
    ...getMockData(method, props as any)!,
    request_id: state.getNextRequestId()
  };

  const event: VKConnectSuccessEvent<K> = {
    detail: {
      type: receiveOnlyMethods.includes(method as any) ? method : method + 'Result',
      data
    }
  };

  return event;
};

export const isReceiveMockMethodExists = (methodName: string): methodName is ReceiveMethodName =>
  methodName in mockDataMap;

export const send: VKConnectSend = async (method, props) => {
  return new Promise((resolve, reject) => {
    if (!isReceiveMockMethodExists(method)) {
      return;
    }

    const event = prepareResponse(method as ReceiveMethodName, props);

    if (!event) {
      return;
    }

    broadcastData(event);
    resolve(event.detail.data as any);
  });
};

export const subscribe = (fn: VKConnectSubscribeHandler) => void state.listeners.push(fn);

export const unsubscribe = (fn: VKConnectSubscribeHandler) => {
  const index = state.listeners.indexOf(fn);

  if (index > -1) {
    state.listeners.splice(index, 1);
  }
};
