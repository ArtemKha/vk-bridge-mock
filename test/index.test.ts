import bridge from '../src';
import { ReceiveMethodName, VKBridgeEvent, RequestIdProp, ReceiveOnlyMethodName } from '@vkontakte/vk-bridge';
import { callReceiveOnlyMethod } from '../src/mockFn';
import { mockDataMap } from '../src/mockData';

export const receiveOnlyMethods: ReceiveOnlyMethodName[] = [
  'VKWebAppAudioPaused',
  'VKWebAppAudioStopped',
  'VKWebAppAudioTrackChanged',
  'VKWebAppAudioUnpaused',
  'VKWebAppInitAds',
  'VKWebAppLoadAds',
  'VKWebAppUpdateConfig',
  'VKWebAppViewHide',
  'VKWebAppViewRestore'
];

export const ioMethods = Object.keys(mockDataMap).filter(methodName => !receiveOnlyMethods.includes(methodName as any));

const dropRequestId = <A extends RequestIdProp>(data: A) => {
  const { request_id, ...out } = data;

  return out;
};

describe('Send', () => {
  ioMethods.forEach(methodName => {
    test(methodName, async () => {
      const mockData = await bridge.send(methodName as any);

      expect(dropRequestId(mockData)).toMatchSnapshot(methodName);
    });
  });
});

describe('Receive only events', () => {
  receiveOnlyMethods.forEach(methodName => {
    test(
      methodName,
      () =>
        new Promise(resolve => {
          const handler = (event: VKBridgeEvent<ReceiveMethodName>) => {
            expect(event.detail.type).toBe(methodName);
            expect(dropRequestId(event.detail.data)).toMatchSnapshot(methodName);

            resolve(bridge.unsubscribe(handler));
          };

          bridge.subscribe(handler);
          callReceiveOnlyMethod(methodName as any);
        })
    );
  });
});
