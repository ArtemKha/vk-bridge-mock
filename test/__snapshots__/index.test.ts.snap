// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receive only events VKWebAppAudioPaused: VKWebAppAudioPaused 1`] = `
Object {
  "id": "audio_track_01.mp3",
  "position": 34,
  "request_id": 99,
  "type": "file",
}
`;

exports[`Receive only events VKWebAppAudioStopped: VKWebAppAudioStopped 1`] = `
Object {
  "request_id": 100,
}
`;

exports[`Receive only events VKWebAppAudioTrackChanged: VKWebAppAudioTrackChanged 1`] = `
Object {
  "id": "audio_track_01.mp3",
  "request_id": 101,
  "type": "file",
}
`;

exports[`Receive only events VKWebAppAudioUnpaused: VKWebAppAudioUnpaused 1`] = `
Object {
  "id": "audio_track_01.mp3",
  "request_id": 102,
  "type": "file",
}
`;

exports[`Receive only events VKWebAppInitAds: VKWebAppInitAds 1`] = `
Object {
  "init": "true",
  "request_id": 103,
}
`;

exports[`Receive only events VKWebAppLoadAds: VKWebAppLoadAds 1`] = `
Object {
  "load": "true",
  "request_id": 104,
}
`;

exports[`Receive only events VKWebAppUpdateConfig: VKWebAppUpdateConfig 1`] = `
Object {
  "app": "vkclient",
  "app_id": "3087106",
  "appearance": "light",
  "insets": Object {
    "bottom": 0,
    "left": 0,
    "right": 0,
    "top": 0,
  },
  "request_id": 105,
  "scheme": "client_light",
  "start_time": 1565272434.911599,
}
`;

exports[`Receive only events VKWebAppViewHide: VKWebAppViewHide 1`] = `
Object {
  "request_id": 106,
}
`;

exports[`Receive only events VKWebAppViewRestore: VKWebAppViewRestore 1`] = `
Object {
  "request_id": 107,
}
`;

exports[`Send promise VKWebAppAddToCommunity: VKWebAppAddToCommunity 1`] = `
Object {
  "group_id": 123456789,
  "request_id": 50,
}
`;

exports[`Send promise VKWebAppAddToFavorites: VKWebAppAddToFavorites 1`] = `
Object {
  "request_id": 91,
  "result": true,
}
`;

exports[`Send promise VKWebAppAllowMessagesFromGroup: VKWebAppAllowMessagesFromGroup 1`] = `
Object {
  "request_id": 51,
  "result": true,
}
`;

exports[`Send promise VKWebAppAllowNotifications: VKWebAppAllowNotifications 1`] = `
Object {
  "enabled": true,
  "request_id": 52,
}
`;

exports[`Send promise VKWebAppCallAPIMethod: VKWebAppCallAPIMethod 1`] = `
Object {
  "request_id": 53,
  "response": Array [
    Object {
      "can_access_closed": true,
      "city": Object {
        "id": 5331,
        "title": "Los Angeles",
      },
      "first_name": "Lindsey",
      "id": 210700286,
      "is_closed": false,
      "last_name": "Stirling",
      "photo_50": "https://sun9-12.u...0uN_RKSXc.jpg?ava=1",
      "verified": 1,
    },
  ],
}
`;

exports[`Send promise VKWebAppClose: VKWebAppClose 1`] = `
Object {
  "payload": Object {},
  "request_id": 55,
}
`;

exports[`Send promise VKWebAppCommunityAccessToken: VKWebAppCommunityAccessToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 95,
}
`;

exports[`Send promise VKWebAppCommunityToken: VKWebAppCommunityToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 96,
}
`;

exports[`Send promise VKWebAppDenyNotifications: VKWebAppDenyNotifications 1`] = `
Object {
  "disabled": true,
  "request_id": 57,
}
`;

exports[`Send promise VKWebAppDisableSwipeBack: VKWebAppDisableSwipeBack 1`] = `
Object {
  "request_id": 97,
}
`;

exports[`Send promise VKWebAppEnableSwipeBack: VKWebAppEnableSwipeBack 1`] = `
Object {
  "request_id": 98,
}
`;

exports[`Send promise VKWebAppFlashGetInfo: VKWebAppFlashGetInfo 1`] = `
Object {
  "is_available": true,
  "level": 1.6,
  "request_id": 58,
}
`;

exports[`Send promise VKWebAppFlashSetLevel: VKWebAppFlashSetLevel 1`] = `
Object {
  "request_id": 59,
  "result": true,
}
`;

exports[`Send promise VKWebAppGetAuthToken: VKWebAppGetAuthToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 54,
  "scope": "friends,photos,video,wall",
}
`;

exports[`Send promise VKWebAppGetClientVersion: VKWebAppGetClientVersion 1`] = `
Object {
  "platform": "android",
  "request_id": 60,
  "version": "5.3.2",
}
`;

exports[`Send promise VKWebAppGetCommunityAuthToken: VKWebAppGetCommunityAuthToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 94,
}
`;

exports[`Send promise VKWebAppGetCommunityToken: VKWebAppGetCommunityToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 93,
}
`;

exports[`Send promise VKWebAppGetEmail: VKWebAppGetEmail 1`] = `
Object {
  "email": "test@gmail.com",
  "request_id": 61,
  "sign": "Y1C99xnbEaR8Wred",
}
`;

exports[`Send promise VKWebAppGetFriends: VKWebAppGetFriends 1`] = `
Object {
  "request_id": 62,
  "users": Array [
    Object {
      "first_name": "Albert",
      "id": 2884043,
      "last_name": "Usmanov",
    },
    Object {
      "first_name": "Aleksandra",
      "id": 1894768,
      "last_name": "Sokolovskaya",
    },
  ],
}
`;

exports[`Send promise VKWebAppGetGeodata: VKWebAppGetGeodata 1`] = `
Object {
  "available": true,
  "lat": "-37",
  "long": "89",
  "request_id": 63,
}
`;

exports[`Send promise VKWebAppGetPersonalCard: VKWebAppGetPersonalCard 1`] = `
Object {
  "address": Object {
    "city": Object {
      "id": 2,
      "name": "Санкт-Петербург",
    },
    "country": Object {
      "id": 1,
      "name": "Россия",
    },
    "postal_code": "191186",
    "specified_address": "Невский пр., д. 28",
  },
  "email": "ivan@gmail.com",
  "phone": "+79001112233",
  "request_id": 64,
}
`;

exports[`Send promise VKWebAppGetPhoneNumber: VKWebAppGetPhoneNumber 1`] = `
Object {
  "is_verified": true,
  "phone_number": "79111234567",
  "request_id": 65,
  "sign": "Y1C99xnbEaR8Wred",
}
`;

exports[`Send promise VKWebAppGetUserInfo: VKWebAppGetUserInfo 1`] = `
Object {
  "bdate": "10.4.1990",
  "city": Object {
    "id": 2,
    "title": "Санкт-Петербург",
  },
  "country": Object {
    "id": 1,
    "title": "Россия",
  },
  "first_name": "Ирина",
  "id": 2314852,
  "last_name": "Денежкина",
  "photo_100": "https://pp.userapi.com/c836333/v836333553/5b138/2eWBOuj5A4g.jpg",
  "photo_200": "https://pp.userapi.com/c836333/v836333553/5b137/tEJNQNigU80.jpg",
  "request_id": 66,
  "sex": 1,
  "timezone": 3,
}
`;

exports[`Send promise VKWebAppJoinGroup: VKWebAppJoinGroup 1`] = `
Object {
  "request_id": 67,
  "result": true,
}
`;

exports[`Send promise VKWebAppOpenApp: VKWebAppOpenApp 1`] = `
Object {
  "request_id": 56,
  "result": true,
}
`;

exports[`Send promise VKWebAppOpenCodeReader: VKWebAppOpenCodeReader 1`] = `
Object {
  "code_data": "Some text",
  "request_id": 68,
}
`;

exports[`Send promise VKWebAppOpenContacts: VKWebAppOpenContacts 1`] = `
Object {
  "first_name": "Сестра",
  "last_name": "",
  "phone": "79217770099",
  "request_id": 70,
}
`;

exports[`Send promise VKWebAppOpenPayForm: VKWebAppOpenPayForm 1`] = `
Object {
  "amount": "1.01",
  "extra": null,
  "request_id": 71,
  "status": true,
  "transaction_id": "150349AC-BDC6-11E9-83B6-5D798FB6FAB1",
}
`;

exports[`Send promise VKWebAppOpenQR: VKWebAppOpenQR 1`] = `
Object {
  "code_data": "Some text",
  "request_id": 69,
}
`;

exports[`Send promise VKWebAppResizeWindow: VKWebAppResizeWindow 1`] = `
Object {
  "height": 1000,
  "request_id": 72,
  "width": 800,
}
`;

exports[`Send promise VKWebAppScroll: VKWebAppScroll 1`] = `
Object {
  "height": 2040,
  "request_id": 73,
  "top": 1,
}
`;

exports[`Send promise VKWebAppSendPayload: VKWebAppSendPayload 1`] = `
Object {
  "request_id": 92,
  "result": true,
}
`;

exports[`Send promise VKWebAppSetLocation: VKWebAppSetLocation 1`] = `
Object {
  "request_id": 74,
  "result": true,
}
`;

exports[`Send promise VKWebAppSetViewSettings: VKWebAppSetViewSettings 1`] = `
Object {
  "request_id": 75,
  "result": true,
}
`;

exports[`Send promise VKWebAppShare: VKWebAppShare 1`] = `
Object {
  "request_id": 76,
  "type": "message",
}
`;

exports[`Send promise VKWebAppShowCommunityWidgetPreviewBox: VKWebAppShowCommunityWidgetPreviewBox 1`] = `
Object {
  "request_id": 77,
  "result": true,
}
`;

exports[`Send promise VKWebAppShowImages: VKWebAppShowImages 1`] = `
Object {
  "request_id": 78,
  "result": true,
}
`;

exports[`Send promise VKWebAppShowInviteBox: VKWebAppShowInviteBox 1`] = `
Object {
  "request_id": 79,
  "success": true,
}
`;

exports[`Send promise VKWebAppShowLeaderBoardBox: VKWebAppShowLeaderBoardBox 1`] = `
Object {
  "request_id": 80,
  "success": true,
}
`;

exports[`Send promise VKWebAppShowMessageBox: VKWebAppShowMessageBox 1`] = `
Object {
  "request_id": 81,
  "result": true,
}
`;

exports[`Send promise VKWebAppShowOrderBox: VKWebAppShowOrderBox 1`] = `
Object {
  "request_id": 82,
  "status": "cancel",
}
`;

exports[`Send promise VKWebAppShowRequestBox: VKWebAppShowRequestBox 1`] = `
Object {
  "requestKey": "123242",
  "request_id": 83,
  "success": true,
}
`;

exports[`Send promise VKWebAppShowWallPostBox: VKWebAppShowWallPostBox 1`] = `
Object {
  "post_id": 1,
  "request_id": 84,
}
`;

exports[`Send promise VKWebAppStorageGet: VKWebAppStorageGet 1`] = `
Object {
  "keys": Array [
    Object {
      "key": "somekey0",
      "value": "somevalue0",
    },
    Object {
      "key": "somekey1",
      "value": "somevalue1",
    },
  ],
  "request_id": 85,
}
`;

exports[`Send promise VKWebAppStorageGetKeys: VKWebAppStorageGetKeys 1`] = `
Object {
  "keys": Array [
    "somekey0",
    "somekey1",
  ],
  "request_id": 86,
}
`;

exports[`Send promise VKWebAppStorageSet: VKWebAppStorageSet 1`] = `
Object {
  "request_id": 87,
  "result": true,
}
`;

exports[`Send promise VKWebAppTapticImpactOccurred: VKWebAppTapticImpactOccurred 1`] = `
Object {
  "request_id": 88,
  "result": true,
}
`;

exports[`Send promise VKWebAppTapticNotificationOccurred: VKWebAppTapticNotificationOccurred 1`] = `
Object {
  "request_id": 89,
  "result": true,
}
`;

exports[`Send promise VKWebAppTapticSelectionChanged: VKWebAppTapticSelectionChanged 1`] = `
Object {
  "request_id": 90,
  "result": true,
}
`;

exports[`Simple send VKWebAppAddToCommunity: VKWebAppAddToCommunity 1`] = `
Object {
  "group_id": 123456789,
  "request_id": 1,
}
`;

exports[`Simple send VKWebAppAddToFavorites: VKWebAppAddToFavorites 1`] = `
Object {
  "request_id": 42,
  "result": true,
}
`;

exports[`Simple send VKWebAppAllowMessagesFromGroup: VKWebAppAllowMessagesFromGroup 1`] = `
Object {
  "request_id": 2,
  "result": true,
}
`;

exports[`Simple send VKWebAppAllowNotifications: VKWebAppAllowNotifications 1`] = `
Object {
  "enabled": true,
  "request_id": 3,
}
`;

exports[`Simple send VKWebAppCallAPIMethod: VKWebAppCallAPIMethod 1`] = `
Object {
  "request_id": 4,
  "response": Array [
    Object {
      "can_access_closed": true,
      "city": Object {
        "id": 5331,
        "title": "Los Angeles",
      },
      "first_name": "Lindsey",
      "id": 210700286,
      "is_closed": false,
      "last_name": "Stirling",
      "photo_50": "https://sun9-12.u...0uN_RKSXc.jpg?ava=1",
      "verified": 1,
    },
  ],
}
`;

exports[`Simple send VKWebAppClose: VKWebAppClose 1`] = `
Object {
  "payload": Object {},
  "request_id": 6,
}
`;

exports[`Simple send VKWebAppCommunityAccessToken: VKWebAppCommunityAccessToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 46,
}
`;

exports[`Simple send VKWebAppCommunityToken: VKWebAppCommunityToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 47,
}
`;

exports[`Simple send VKWebAppDenyNotifications: VKWebAppDenyNotifications 1`] = `
Object {
  "disabled": true,
  "request_id": 8,
}
`;

exports[`Simple send VKWebAppDisableSwipeBack: VKWebAppDisableSwipeBack 1`] = `
Object {
  "request_id": 48,
}
`;

exports[`Simple send VKWebAppEnableSwipeBack: VKWebAppEnableSwipeBack 1`] = `
Object {
  "request_id": 49,
}
`;

exports[`Simple send VKWebAppFlashGetInfo: VKWebAppFlashGetInfo 1`] = `
Object {
  "is_available": true,
  "level": 1.6,
  "request_id": 9,
}
`;

exports[`Simple send VKWebAppFlashSetLevel: VKWebAppFlashSetLevel 1`] = `
Object {
  "request_id": 10,
  "result": true,
}
`;

exports[`Simple send VKWebAppGetAuthToken: VKWebAppGetAuthToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 5,
  "scope": "friends,photos,video,wall",
}
`;

exports[`Simple send VKWebAppGetClientVersion: VKWebAppGetClientVersion 1`] = `
Object {
  "platform": "android",
  "request_id": 11,
  "version": "5.3.2",
}
`;

exports[`Simple send VKWebAppGetCommunityAuthToken: VKWebAppGetCommunityAuthToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 45,
}
`;

exports[`Simple send VKWebAppGetCommunityToken: VKWebAppGetCommunityToken 1`] = `
Object {
  "access_token": "yours000access111token222yours000access111token222yours000access111token2220123456789",
  "request_id": 44,
}
`;

exports[`Simple send VKWebAppGetEmail: VKWebAppGetEmail 1`] = `
Object {
  "email": "test@gmail.com",
  "request_id": 12,
  "sign": "Y1C99xnbEaR8Wred",
}
`;

exports[`Simple send VKWebAppGetFriends: VKWebAppGetFriends 1`] = `
Object {
  "request_id": 13,
  "users": Array [
    Object {
      "first_name": "Albert",
      "id": 2884043,
      "last_name": "Usmanov",
    },
    Object {
      "first_name": "Aleksandra",
      "id": 1894768,
      "last_name": "Sokolovskaya",
    },
  ],
}
`;

exports[`Simple send VKWebAppGetGeodata: VKWebAppGetGeodata 1`] = `
Object {
  "available": true,
  "lat": "-37",
  "long": "89",
  "request_id": 14,
}
`;

exports[`Simple send VKWebAppGetPersonalCard: VKWebAppGetPersonalCard 1`] = `
Object {
  "address": Object {
    "city": Object {
      "id": 2,
      "name": "Санкт-Петербург",
    },
    "country": Object {
      "id": 1,
      "name": "Россия",
    },
    "postal_code": "191186",
    "specified_address": "Невский пр., д. 28",
  },
  "email": "ivan@gmail.com",
  "phone": "+79001112233",
  "request_id": 15,
}
`;

exports[`Simple send VKWebAppGetPhoneNumber: VKWebAppGetPhoneNumber 1`] = `
Object {
  "is_verified": true,
  "phone_number": "79111234567",
  "request_id": 16,
  "sign": "Y1C99xnbEaR8Wred",
}
`;

exports[`Simple send VKWebAppGetUserInfo: VKWebAppGetUserInfo 1`] = `
Object {
  "bdate": "10.4.1990",
  "city": Object {
    "id": 2,
    "title": "Санкт-Петербург",
  },
  "country": Object {
    "id": 1,
    "title": "Россия",
  },
  "first_name": "Ирина",
  "id": 2314852,
  "last_name": "Денежкина",
  "photo_100": "https://pp.userapi.com/c836333/v836333553/5b138/2eWBOuj5A4g.jpg",
  "photo_200": "https://pp.userapi.com/c836333/v836333553/5b137/tEJNQNigU80.jpg",
  "request_id": 17,
  "sex": 1,
  "timezone": 3,
}
`;

exports[`Simple send VKWebAppJoinGroup: VKWebAppJoinGroup 1`] = `
Object {
  "request_id": 18,
  "result": true,
}
`;

exports[`Simple send VKWebAppOpenApp: VKWebAppOpenApp 1`] = `
Object {
  "request_id": 7,
  "result": true,
}
`;

exports[`Simple send VKWebAppOpenCodeReader: VKWebAppOpenCodeReader 1`] = `
Object {
  "code_data": "Some text",
  "request_id": 19,
}
`;

exports[`Simple send VKWebAppOpenContacts: VKWebAppOpenContacts 1`] = `
Object {
  "first_name": "Сестра",
  "last_name": "",
  "phone": "79217770099",
  "request_id": 21,
}
`;

exports[`Simple send VKWebAppOpenPayForm: VKWebAppOpenPayForm 1`] = `
Object {
  "amount": "1.01",
  "extra": null,
  "request_id": 22,
  "status": true,
  "transaction_id": "150349AC-BDC6-11E9-83B6-5D798FB6FAB1",
}
`;

exports[`Simple send VKWebAppOpenQR: VKWebAppOpenQR 1`] = `
Object {
  "code_data": "Some text",
  "request_id": 20,
}
`;

exports[`Simple send VKWebAppResizeWindow: VKWebAppResizeWindow 1`] = `
Object {
  "height": 1000,
  "request_id": 23,
  "width": 800,
}
`;

exports[`Simple send VKWebAppScroll: VKWebAppScroll 1`] = `
Object {
  "height": 2040,
  "request_id": 24,
  "top": 1,
}
`;

exports[`Simple send VKWebAppSendPayload: VKWebAppSendPayload 1`] = `
Object {
  "request_id": 43,
  "result": true,
}
`;

exports[`Simple send VKWebAppSetLocation: VKWebAppSetLocation 1`] = `
Object {
  "request_id": 25,
  "result": true,
}
`;

exports[`Simple send VKWebAppSetViewSettings: VKWebAppSetViewSettings 1`] = `
Object {
  "request_id": 26,
  "result": true,
}
`;

exports[`Simple send VKWebAppShare: VKWebAppShare 1`] = `
Object {
  "request_id": 27,
  "type": "message",
}
`;

exports[`Simple send VKWebAppShowCommunityWidgetPreviewBox: VKWebAppShowCommunityWidgetPreviewBox 1`] = `
Object {
  "request_id": 28,
  "result": true,
}
`;

exports[`Simple send VKWebAppShowImages: VKWebAppShowImages 1`] = `
Object {
  "request_id": 29,
  "result": true,
}
`;

exports[`Simple send VKWebAppShowInviteBox: VKWebAppShowInviteBox 1`] = `
Object {
  "request_id": 30,
  "success": true,
}
`;

exports[`Simple send VKWebAppShowLeaderBoardBox: VKWebAppShowLeaderBoardBox 1`] = `
Object {
  "request_id": 31,
  "success": true,
}
`;

exports[`Simple send VKWebAppShowMessageBox: VKWebAppShowMessageBox 1`] = `
Object {
  "request_id": 32,
  "result": true,
}
`;

exports[`Simple send VKWebAppShowOrderBox: VKWebAppShowOrderBox 1`] = `
Object {
  "request_id": 33,
  "status": "cancel",
}
`;

exports[`Simple send VKWebAppShowRequestBox: VKWebAppShowRequestBox 1`] = `
Object {
  "requestKey": "123242",
  "request_id": 34,
  "success": true,
}
`;

exports[`Simple send VKWebAppShowWallPostBox: VKWebAppShowWallPostBox 1`] = `
Object {
  "post_id": 1,
  "request_id": 35,
}
`;

exports[`Simple send VKWebAppStorageGet: VKWebAppStorageGet 1`] = `
Object {
  "keys": Array [
    Object {
      "key": "somekey0",
      "value": "somevalue0",
    },
    Object {
      "key": "somekey1",
      "value": "somevalue1",
    },
  ],
  "request_id": 36,
}
`;

exports[`Simple send VKWebAppStorageGetKeys: VKWebAppStorageGetKeys 1`] = `
Object {
  "keys": Array [
    "somekey0",
    "somekey1",
  ],
  "request_id": 37,
}
`;

exports[`Simple send VKWebAppStorageSet: VKWebAppStorageSet 1`] = `
Object {
  "request_id": 38,
  "result": true,
}
`;

exports[`Simple send VKWebAppTapticImpactOccurred: VKWebAppTapticImpactOccurred 1`] = `
Object {
  "request_id": 39,
  "result": true,
}
`;

exports[`Simple send VKWebAppTapticNotificationOccurred: VKWebAppTapticNotificationOccurred 1`] = `
Object {
  "request_id": 40,
  "result": true,
}
`;

exports[`Simple send VKWebAppTapticSelectionChanged: VKWebAppTapticSelectionChanged 1`] = `
Object {
  "request_id": 41,
  "result": true,
}
`;
